### PR TITLE
Feature: processing data in models

### PR DESCRIPTION
**Pull request by nachosala89**
- Created model classes for all entities in ERD.
- Set up associations between models.
- Included additional methods.
- Created current_user method in ApplicationController.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
-  belongs_to :author_id
+  belongs_to :author, class_name: 'User'
+  has_many :comment
+  has_many :like
+
+  def update_counter(user_id = author_id)
+    user = User.find(user_id)
+    user.posts_counter = Post.where(author_id: user.id).count
+    user.save
+  end
```
Comment (petudeveloper) [inline app/models/post.rb]: - [ ] Your `update_counter` method for the Post model gets the user's posts but is not updating(incrementing) the posts counter. Please add the necessary logic to fix this.
For instance, you can try using `user.increment!(:posts_counter)` or `user.posts_counter += 1`

[OPTIONAL] You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private

```
  # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    # logic here
  end

```

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,4 +1,10 @@
 class Comment < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User'
   belongs_to :post
+
+  def update_counter(current_post_id = post_id)
+    post = Post.find(current_post_id)
+    post.comments_counter = Comment.where(post_id: post.id).count
+    post.save
+  end
```
Comment (petudeveloper) [inline app/models/comment.rb]: - [ ] Your `update_counter` method for the Comment model gets the post's comments but is not updating(incrementing) the comments counter. Please add the necessary logic to fix this.
For instance, you can try using `post.increment!(:comments_counter)` or `post.comments_counter += 1`

[OPTIONAL] You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_comments_counter` method every time a new post has been saved to database, then make the `update_comments_counter` method private

```
  # app/models/comment.rb
  after_save :update_comments_counter

  private

  def update_comments_counter
    # logic here
  end

```

**Review by petudeveloper — CHANGES_REQUESTED** [review]
Hi @nachosala89 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
very close


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/models/post.rb (hunk 1 above)
Comment (nachosala89) [reply app/models/post.rb]: Hi @petudeveloper. Thank you for your suggestions! But my method already returns the number of posts a user has. It uses the `count` method, to count every post where `author_id: user.id`. Why should I change that?

File: app/models/comment.rb (hunk 2 above)
Comment (nachosala89) [reply app/models/comment.rb]: And it's the same for this.

**Review by dasileker — APPROVED** [review]
## **Hi @nachosala89,** 

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 

Approved

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**